trie/trieStorageManager.go

```diff
@@ -115,6 +120,21 @@ func (tsm *trieStorageManager) storageProcessLoop(ctx context.Context, msh marsh
 	}
 }
 
+func (tsm *trieStorageManager) cleanupChans() {
+	<-tsm.closer.ChanClose()
+	//at this point we can not add new entries in the snapshot/checkpoint chans
+	for {
+		select {
+		case entry := <-tsm.snapshotReq:
+			tsm.finishOperation(entry, "trie snapshot finished on cleanup")
+		case entry := <-tsm.checkpointReq:
+			tsm.finishOperation(entry, "trie checkpoint finished on cleanup")
+		default:
```

Comment:
maybe add a timeout? from my understanding, it will continuously process new entries, but the entire process doesn't have a feedback when it finished

Reply:
No, the component won't add new entries in the chans, because of L284-L289 and L314-L319. No need for the timeout as this function will need to empty all channels and then, when all channels are empty it will take the default case of the select.